cmd/agent/app/run.go

```diff
@@ -374,7 +374,9 @@ func StartAgent() error {
 	}
 
 	// start dependent services
-	startDependentServices()
+	go func () {
+		startDependentServices()
+	}()
```

Comment:
Since this now runs in a separate goroutine, there's a chance that the dependent services will actually be started only after the main loop in the service's `Execute` function exits and the agent service initiates a shutdown. Is there a separate mechanism preventing this scenario from happening? If not, do you know what the impact of such a scenario would be?

Unrelated, I think this can be simplified to:

```suggestion
	go startDependentServices()
```

Reply:
what _should_ happen in that case is that the SCM should stop the dependent services, b/c they're dependent on the core agent and the core agent is no longer running.
If that mechanism doesn't work right, we could end up back where we already are, with (e.g.) trace agent running but core agent not.

Reply:
ok, thanks